cmd/client-s3.go

```diff
@@ -864,71 +851,75 @@ func (c *s3Client) Stat(isIncomplete bool) (*clientContent, *probe.Error) {
 	if isIncomplete {
 		for objectMultipartInfo := range c.api.ListIncompleteUploads(bucket, object, nonRecursive, nil) {
 			if objectMultipartInfo.Err != nil {
-				return nil, probe.NewError(objectMultipartInfo.Err)
+				return nil, nil, probe.NewError(objectMultipartInfo.Err)
 			}
 
 			if objectMultipartInfo.Key == object {
 				objectMetadata.URL = *c.targetURL
 				objectMetadata.Time = objectMultipartInfo.Initiated
 				objectMetadata.Size = objectMultipartInfo.Size
 				objectMetadata.Type = os.FileMode(0664)
-				return objectMetadata, nil
+				return objectMetadata, nil, nil
 			}
 
 			if strings.HasSuffix(objectMultipartInfo.Key, string(c.targetURL.Separator)) {
 				objectMetadata.URL = *c.targetURL
 				objectMetadata.Type = os.ModeDir
-				return objectMetadata, nil
+				return objectMetadata, nil, nil
 			}
 		}
-		return nil, probe.NewError(ObjectMissing{})
+		return nil, nil, probe.NewError(ObjectMissing{})
 	}
 
 	for objectStat := range c.listObjectWrapper(bucket, object, nonRecursive, nil) {
 		if objectStat.Err != nil {
-			return nil, probe.NewError(objectStat.Err)
+			return nil, nil, probe.NewError(objectStat.Err)
 		}
 
 		if objectStat.Key == object {
 			objectMetadata.URL = *c.targetURL
 			objectMetadata.Time = objectStat.LastModified
 			objectMetadata.Size = objectStat.Size
 			objectMetadata.Type = os.FileMode(0664)
-			return objectMetadata, nil
+			return objectMetadata, map[string][]string{}, nil
 		}
 
 		if strings.HasSuffix(objectStat.Key, string(c.targetURL.Separator)) {
 			objectMetadata.URL = *c.targetURL
 			objectMetadata.Type = os.ModeDir
-			return objectMetadata, nil
+			return objectMetadata, map[string][]string{}, nil
 		}
 	}
 	objectStat, e := c.api.StatObject(bucket, object)
 	if e != nil {
 		errResponse := minio.ToErrorResponse(e)
 		if errResponse.Code == "AccessDenied" {
-			return nil, probe.NewError(PathInsufficientPermission{Path: c.targetURL.String()})
+			return nil, nil, probe.NewError(PathInsufficientPermission{Path: c.targetURL.String()})
 		}
 		if errResponse.Code == "NoSuchBucket" {
-			return nil, probe.NewError(BucketDoesNotExist{
+			return nil, nil, probe.NewError(BucketDoesNotExist{
 				Bucket: bucket,
 			})
 		}
 		if errResponse.Code == "InvalidBucketName" {
-			return nil, probe.NewError(BucketInvalid{
+			return nil, nil, probe.NewError(BucketInvalid{
 				Bucket: bucket,
 			})
 		}
 		if errResponse.Code == "NoSuchKey" || errResponse.Code == "InvalidArgument" {
-			return nil, probe.NewError(ObjectMissing{})
+			return nil, nil, probe.NewError(ObjectMissing{})
 		}
-		return nil, probe.NewError(e)
+		return nil, nil, probe.NewError(e)
 	}
 	objectMetadata.URL = *c.targetURL
 	objectMetadata.Time = objectStat.LastModified
 	objectMetadata.Size = objectStat.Size
 	objectMetadata.Type = os.FileMode(0664)
-	return objectMetadata, nil
+
+	metadata := objectStat.Metadata
```

Comment:
> Yes. IIUC, we don't use metadata everywhere. So, I am suggesting we construct map[string] string wherever required with corresponding values found in objectMetadata from the Stat method. Does that make sense?

Yes, I agree.

So the client interface will be changed from:
```
 40 type Client interface {
 41         // Common operations
 42         Stat(isIncomplete bool) (content *clientContent, metadata map[string][]string, err *probe.Error)
```
To
```
 40 type Client interface {
 41         // Common operations
 42         Stat(isIncomplete bool) (content *clientContent, err *probe.Error)
```
and clientContent will have new Metadata additional field:
```
 75 type clientContent struct {
 76         URL  clientURL
 77         Time time.Time
 78         Size int64
 79         Type os.FileMode
            Metadata map[string][]string
 80         Err  *probe.Error
 81 }
```

@abperiasamy, @harshavardhana what do you think ?